### PR TITLE
feat(hub-pool): Reduce size of state vars

### PR DESCRIPTION
**Pull request by nicholaspai**

### CODE REVIEW

File: contracts/HubPoolInterface.sol [hunk 1]
```diff
@@ -26,6 +21,11 @@ interface HubPoolInterface {
         // A positive number indicates that the HubPool owes the SpokePool funds. A negative number indicates that the
         // SpokePool owes the HubPool funds. See the comment above for the dynamics of this and netSendAmounts
         int256[] runningBalances;
+        // Used as the index in the bitmap to track whether this leaf has been executed or not.
+        uint32 leafId;
```
Comment (nicholaspai) [inline contracts/HubPoolInterface.sol:25]: Should this be the same size as the `poolRebalanceLeafCount` in the HubPool functions, aka uint8?

File: contracts/HubPool.sol [hunk 2]
```diff
@@ -126,26 +126,26 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
     event WhitelistRoute(uint256 destinationChainId, address originToken, address destinationToken);
 
     event InitiateRefundRequested(
-        uint64 requestExpirationTimestamp,
-        uint64 unclaimedPoolRebalanceLeafCount,
         uint256[] bundleEvaluationBlockNumbers,
+        uint32 requestExpirationTimestamp,
```
Comment (mrice32) [inline contracts/HubPool.sol:130]: As discussed IRL, these won't get packed no matter the order, but it's totally fine to move them for style!

**Review by mrice32 — APPROVED** [review]
LGTM!

File: contracts/HubPoolInterface.sol (hunk 1 above)
Comment (chrismaree) [reply contracts/HubPoolInterface.sol:25]: I think yes! as long as the leafId in the relayer refund remains large we are good.

**Review by chrismaree — APPROVED** [review]

File: contracts/HubPool.sol [hunk 3]
```diff
@@ -29,30 +29,47 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
     using Address for address;
 
     struct RefundRequest {
-        uint64 requestExpirationTimestamp;
-        uint64 unclaimedPoolRebalanceLeafCount;
+        uint256 claimedBitMap; // This is a 1D bitmap, with max size of 256 elements, limiting us to 256 chainsIds.
+        uint32 requestExpirationTimestamp;
+        uint8 unclaimedPoolRebalanceLeafCount;
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply contracts/HubPool.sol:34]: address and bool can be packed with small uints..?

**Pull request merged**